### PR TITLE
APIM-490: Support of metrics/logging on api v4

### PR DESCRIPTION
**Pull request by guillaumelamirand**
**Issue**

https://gravitee.atlassian.net/browse/APIM-490

**Additional context**

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.1.0-support-v4-logging-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/gateway/gravitee-gateway-api/2.1.0-support-v4-logging-SNAPSHOT/gravitee-gateway-api-2.1.0-support-v4-logging-SNAPSHOT.zip)

### CODE REVIEW

File: src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java [hunk 1]
```diff
@@ -21,6 +21,91 @@
...
+    /**
+     * Get the current id of the message.
+     * The id may be <code>null</code>.
+     *
+     * @return the id of the message or <code>null</code> in case the message doesn't have any id.
+     */
+    String id();
+
+    String correlationId();
+    String parentCorrelationId();
```
Comment (jgiovaresco) [inline src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java:33]: Should we add a javadoc explaining why those ids were defined? If I remember correctly, this id helps to link messages when a message is created from another message

File: src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java [hunk 2]
```diff
@@ -21,6 +21,91 @@
...
+     *
+     * @param content the buffer representing the content of the message.
+     * @return reference to itself for easily chain calls.
+     */
+    Message content(final String content);
+
+    /**
+     * Allow acknowledging this message when it has been well-processed.
+     */
+    void ack();
```
Comment (jgiovaresco) [inline src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java:107]: What about returning the message like `content` setter to easily chain calls?

**Review by jgiovaresco — APPROVED** [review]
The cycle dependency btw gateway-api and reporter-api does not help the CI 😞

File: src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java (hunk 1 above)
Comment (guillaumelamirand) [reply src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java:33]: of course, PR was in draft ;)

File: src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java (hunk 2 above)
Comment (jgiovaresco) [reply src/main/java/io/gravitee/gateway/jupiter/api/message/Message.java:119]: I don't know it is just for consistency 🙈

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**